source/D2Common/src/Path/PathMisc.cpp

```diff
@@ -878,6 +878,7 @@ TanToDirectionLutEntryStrc lutTanToDirection[] = {
 static_assert(ARRAY_SIZE(lutTanToDirection) == 128, "There should be 128 entries in this lookup table");
 
 //D2Common.0x6FDAC5E0
+// Should be in Step.cpp
```

Comment:
What's the reasoning for `PATH_GetDirectionVector_6FDAC5E0` being in Step.cpp ?

Reply:
In 1.00-1.09, there is an assertion to make sure the tangent index doesn't go out of bounds:
```c
nIndex = (int)(127 * (v8 - v7)) / (int)(v10 - v9);
if ( nIndex >= 128 )
{
	Fog_10021("nIndex < STEP_ARRAY_LENGTH", "C:\\D2\\Source\\D2Common\\PATH\\Step.cpp", 341);
	exit(-1);
}
```
1.00-1.03 were compiled with Visual C++6 service pack 3, and I believe 1.04+ is service pack 5. They may have removed some asserts in 1.10, but regardless I know compiler optimizations when I see them, and 1.10 absolutely has more garbled assembly than 1.09 lol. (Seems unlikely they'd move the function out of Step.cpp)

Reply:
Oh cool, there are indeed more asserts depending on the versions! I rarely check those unless I am looking for a specific string, but I guess this could help to finish reversing some other PATH functions. (1.13c already helps a lot as the compiler was less aggressive with the inlining).